### DISCUSSION TITLE
Datatable with hierarchy

### DISCUSSION THREAD

**Discussion by misterek** [Textualize/textual#3067] [Q&A]
I have a set of information that I'd like to display, but there also is a hierarchy component to it as well.

Think of it like
car, weight, price
carpart, weight, price
smallercarpart, weight,price
(not the best example :) )

So, they all make sense in the same table, but there's also a hierarchy.  I've been managing it by adding/removing rows to the table, but sorting them is an annoying challenge (essentially I use a hidden column to sort and maintain the hierarchy)

I'm aware of tree, but that doesn't work so well when you have a table-like structure.

Am I missing anything obvious that would help in this situation?

**Comment by TomJGooding** [comment]
Sorry, perhaps I'm missing something but it's not quite clear what you mean by a table hierarchy?

Do you mean a way of sorting data by multiple columns, the equivalent of a [Custom Sort in Excel](https://support.microsoft.com/en-us/office/sort-data-in-a-range-or-table-62d0b95d-2a90-4610-a6ae-2e545c4a4654)?

**Comment by misterek** [comment]
Apologies.  I described it really poorly.  This looks like a screenshot that more describes what I'm trying to do:
https://images.g2crowd.com/uploads/attachment/file/127851/iteration-status.png (From an old piece of agile management software).

Note their is a hierarchy there (tasks are children of user stories).  However, the vast majority of the table is the same for tasks or user stories.  So, in that example, you click on a user story and the tasks are expanded, but the columns of the table are the same.

So, it's kind of a weird merging of a DataTable and a Tree.

**Reply by TomJGooding** [reply]
Thanks, I understand now from that example screenshot!

I would probably have the same approach and add/remove rows to make the table 'expandable'. I'm not sure there is an easier way of achieving this unfortunately, but others much smarter than me in the Textual community might have some ideas!
